main/src/library/MutDeque.flix

```diff
@@ -185,28 +181,38 @@ namespace MutDeque {
     ///
     def shrink!(d: MutDeque[a]): Unit & Impure =
         let MutDeque(a, f, b) = d;
-        let mc = minCapacity();
         let c = capacity(d);
-        if (c > mc) {                                      // Prevent the backing array from shrinking below `minCapacity`.
-            let arr = [default; c / 2];                    // Allocate new array `arr` with half the capacity of `a`.
-            copyElements!(deref f, deref b, deref a, arr); // Copy elements from old array `a` to new array `arr`.
-            a := arr;                                      // Update references.
-            b := computeSize(c, deref f, deref b);
-            f := 0
+        if (c > minCapacity()) { // Prevent the backing array from shrinking below `minCapacity`.
+            copyElements!(f, b, a, c / 2) // Copy elements into new array and update references.
         } else {
             ()
         }
 
     ///
     /// Copies the elements from `a` to `a1`. Mutates the array `a1`.
     ///
-    def copyElements!(f: Int32, b: Int32, a: Array[a], a1: Array[a]): Unit & Impure =
-        let c = Array.length(a);
-        if (f < b) { // If this predicate is true the elements do not "wrap around" in the array, i.e. the elements are laid out sequentially from [0 .. b].
-            Array.updateSequence!(0,     a[f .. b], a1)
+    def copyElements!(f: Ref[Int32], b: Ref[Int32], a: Ref[Array[a]], c1: Int32): Unit & Impure =
+        let f1 = deref f;
```

Comment:
Why so complicated? 

Assume we have from `begin` to `end`. Assume the length is `l` and we want to double to `2 x l`.

Could we not just `copyRange(begin, 2 x l, arr`)?

The deque would then start from `0` and go to `end - begin` or something like that.